libecho/src/IReceiver.h

```diff
@@ -0,0 +1,5 @@
+class IReceiver {
```

Comment:
Don't you need virtual destructor? Also, every receiver needs, for example, windowSize, so maybe add a generic constructor to simplify construction for derived class?